src/tactic/norm_num.lean

```diff
@@ -36,89 +36,508 @@ variable {α : Type u}
 lemma subst_into_neg {α} [has_neg α] (a ta t : α) (pra : a = ta) (prt : -ta = t) : -a = t :=
 by simp [pra, prt]
 
-theorem bit0_zero [add_group α] : bit0 (0 : α) = 0 := add_zero _
+meta inductive match_numeral_result
+| zero | one | bit0 (e : expr) | bit1 (e : expr)
```

Comment:
```suggestion
| zero | one | bit0 (e : expr) | bit1 (e : expr) | other
```

Then you don't need to wrap it in an `option`, and you can save a few instructions.